pkg/controller/controller_utils.go

```diff
@@ -629,6 +670,86 @@ func maxContainerRestarts(pod *api.Pod) int {
 	return maxRestarts
 }
 
+// ControlActivePods first filters out inactive pods, then it tries to take
+// control of the remaining pods that matching the selector. It returns the pods
+// that are active and are controlled by the controller, and the pods that
+// should be abandoned by the controller.
+func ControlActivePods(podControl PodControlInterface,
```

Comment:
nit: can you group all the control* methods into a type that exposes a single public method (ControlActivePods) and make everything else private?